### ISSUE TITLE
Double initialisation?

### ISSUE THREAD

**Issue by alexbainbridge** [ionic-team/ionic-storage#54]
On 1.1.7 (and earlier), in the browser console logs, I aways see 2 of

Ionic Storage driver: asyncStorage

This is not broken as such, in that it all works, but perhaps the code is initialising twice? In the context of finding performance gains, I wonder if this something that could be looked at?

Incidentally, this error message (separate issue), was also seen twice
https://github.com/driftyco/ionic-storage/issues/53

**Comment by astec** [comment]
I see same duplicate log records.

**Comment by chron0** [comment]
Dito, and, although not yet confirmed with empirical tests, I'd also agree on the sentiment that ionic-storage pulls heavy on startup time. When I used native localstorage before, it seemed way faster. But since localstorage is not an option for certain platforms/deployments it would be great to see ionic-storage perform at a similar level (ionickv/indexdb).

**Comment by chron0** [comment]
There is some other oddity I've discovered when setting a different db/store name (other than the default _ionickv), which may have the same origin:

```
export function provideStorage(): any {
  return new Storage(
    [
      'sqlite',
      'indexeddb',
      'websql',
      'localstorage',
    ],
    {
      name: 'governessdb',
      storeName: 'governessdata',
    }
  );
}

[...]

export function providers(): any {
  return [
    [...]
    // Enable Ionic's storage engine
    { provide: Storage, useFactory: provideStorage },
  ];
}


@NgModule({
  [...]
  providers: providers(),
})
```

It still creates the _ionickv indexeddb and now half of the data is in one, while other stuff, which was triggered later within the application is in governessdata.

![screenshot_2016-12-29_15-23-53](https://cloud.githubusercontent.com/assets/3270028/21546960/d8291a94-cdda-11e6-89cf-24aca4e86153.png)

**Comment by mlynch** [comment]
Hmm, not able to reproduce

**Comment by chron0** [comment]
@mlynch: Thanks for the reminder. Since you couldn't reproduce I went ahead and tried to figure it out. I haven't done a second round to re-verify but in my case it seemed to have been caused by wrapping ionic-storage into a storage provider/service and then using this in the actual data provider services. 

Before there was ionic-storage I created my own storage provider to wrap native localstorage and after ionic-storage was released I just swapped my localstorage calls with ionic-storage ones.

After importing and using ionic-storage in each data provider/service instead of my own storage provider it creates only the governessdata storage instance. I have no idea why it should make a difference but it seems to do.
